sound/soc/sof/ipc.c

```diff
@@ -286,10 +286,16 @@ static int sof_ipc_tx_message_unlocked(struct snd_sof_ipc *ipc, u32 header,
 				       void *msg_data, size_t msg_bytes,
 				       void *reply_data, size_t reply_bytes)
 {
+	struct sof_ipc_cmd_hdr *hdr = msg_data;
```

Comment:
I find the naming conventions complicated:

````
 u32 header
struct sof_ipc_cmd_hdr *hdr
````

what's the difference between the two variables?

Reply:
@plbossart, the `struct sof_ipc_cmd_hdr *hdr` is the IPC message header containing `size` and `cmd`
The `u32 header` parameter _is_ the `hdr->cmd`

Throughout the code we use `struct sof_ipc_cmd_hdr *hdr` and I don't want to diverge from it, the `header` on the other hand just lost it's only user in this series (for IPC3, that is).

We were saving the `header` to `msg->header` which is s`struct snd_sof_ipc_msg` and it acts as a means of passing the message from core to platform code to do the transfer.